lib/src/rules/pattern.ts

```diff
@@ -0,0 +1,34 @@
+import { FieldValidationResult } from '../entities';
+
+export interface PatternParams {
+  pattern: string | RegExp;
+}
+
+const BAD_PARAMETER = 'FieldValidationError: pattern option for pattern validation is mandatory. Example: { pattern: /\d+/ }.';
+export function pattern(value: string, vm, customParams: PatternParams): FieldValidationResult {
+  const pattern = parsePattern(customParams);
+  const isValid = pattern.test(value);
+  const validationResult = new FieldValidationResult();
+  validationResult.succeeded = isValid;
+  validationResult.type = 'PATTERN';
+  validationResult.errorMessage = isValid ? '' : `Please provide a valid format.`;
+
+  return validationResult;
+}
+
+function parsePattern({ pattern }: PatternParams): RegExp {
+  if (pattern === undefined || pattern === null) {
```

Comment:
That's not true for `""`, `false` or `0` values. I tried to avoid RegExp like `/null/` I'll change that to:

```ts
if (typeof pattern === 'boolean' || pattern === null) {
    throw new Error(BAD_PARAMETER);
  }
  return getRegExp(pattern);
```

to avoid `/true/` and `/false/`  too (with a comment).